src/PlusServer/Commands/vtkPlusWinProbeCommand.cxx

```diff
@@ -0,0 +1,942 @@
+  for (paramIt = this->RequestedParameters.begin(); paramIt != this->RequestedParameters.end(); ++paramIt)
+  {
+    vtkSmartPointer<vtkXMLDataElement> paramElem = vtkSmartPointer<vtkXMLDataElement>::New();
+    paramElem->SetName("Parameter");
+    paramElem->SetAttribute("Name", paramIt->first.c_str());
+    std::map<std::string, std::string> param_attributes = paramIt->second;
+    std::map<std::string, std::string>::iterator attribIt;
+    for(attribIt = param_attributes.begin(); attribIt != param_attributes.end(); ++attribIt)
+    {
+      paramElem->SetAttribute(attribIt->first.c_str(), attribIt->second.c_str());
+    }
+    aConfig->AddNestedElement(paramElem);
+  }
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusWinProbeCommand::Execute()
+{
+  LOG_DEBUG("vtkPlusWinProbeCommand::Execute: " << (!this->CommandName.empty() ? this->CommandName : "(undefined)")
+            << ", device: " << (this->DeviceId.empty() ? "(undefined)" : this->DeviceId)
+            << ", value: " << (this->CommandValue.empty() ? "(undefined)" : this->CommandValue)
+            << ", index: " << (this->CommandIndex.empty() ? "(undefined)" : this->CommandIndex));
+
+  // Data Collector
+  vtkPlusDataCollector* dataCollector = GetDataCollector();
+  if (dataCollector == NULL)
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", "Invalid data collector.");
+    return PLUS_FAIL;
+  }
+
+  // Get device pointer
+  if (this->DeviceId.empty())
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", "No DeviceId specified.");
+    return PLUS_FAIL;
+  }
+  vtkPlusDevice* usDevice = NULL;
+  if (dataCollector->GetDevice(usDevice, this->DeviceId) != PLUS_SUCCESS)
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", std::string("Device ")
+                               + (this->DeviceId.empty() ? "(undefined)" : this->DeviceId) + std::string(" is not found."));
+    return PLUS_FAIL;
+  }
+  vtkPlusWinProbeVideoSource* device = dynamic_cast<vtkPlusWinProbeVideoSource*>(usDevice);
+
+  // CommandName
+  if(!igsioCommon::IsEqualInsensitive(this->Name, WINPROBE_CMD))
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", "Unknown wee command name: " + this->Name + ".");
+    return PLUS_FAIL;
+  }
+
+  std::map < std::string, std::pair<IANA_ENCODING_TYPE, std::string> > metaData;
+  std::string resultString = "<CommandReply>";
+  std::string error = "";
+  std::string res;
+  PlusStatus status = PLUS_SUCCESS;
+  bool hasFailure = false;
+
+  std::list<std::pair<std::string, std::map<std::string, std::string>>>::iterator paramIt;
+  for (paramIt = this->RequestedParameters.begin(); paramIt != this->RequestedParameters.end(); ++paramIt)
+  {
+    std::string parameterName = paramIt->first;
+    std::map<std::string, std::string> attribs = paramIt->second;
```

Comment:
What are the two strings being mapped here?  Can you put a comment in the code here with the description.

Reply:
updated